src/helpers/trezorTransformTransaction.ts

```diff
@@ -0,0 +1,203 @@
+// https://github.com/trezor/connect/blob/develop/src/js/plugins/stellar/plugin.js
+import StellarSdk, { MemoType, MemoValue, Asset } from "stellar-sdk";
+import BigNumber from "bignumber.js";
+
+/**
+ * Transforms StellarSdk.Signer to TrezorConnect.StellarTransaction.Signer
+ */
+const transformSigner = (signer: {
+  ed25519PublicKey?: string;
+  sha256Hash?: string | Buffer;
+  preAuthTx?: string | Buffer;
+  weight?: number | string;
+}) => {
+  let type = 0;
+  let key: any;
+  const { weight } = signer;
+  if (typeof signer.ed25519PublicKey === "string") {
+    const keyPair = StellarSdk.Keypair.fromPublicKey(signer.ed25519PublicKey);
+    key = keyPair.rawPublicKey().toString("hex");
+  }
+  if (signer.preAuthTx instanceof Buffer) {
+    type = 1;
+    key = signer.preAuthTx.toString("hex");
+  }
+  if (signer.sha256Hash instanceof Buffer) {
+    type = 2;
+    key = signer.sha256Hash.toString("hex");
+  }
+  return {
+    type,
+    key,
+    weight,
+  };
+};
+
+/**
+ * Transforms StellarSdk.Asset to TrezorConnect.StellarTransaction.Asset
+ */
+const transformAsset = (asset: Asset) => {
+  if (asset.isNative()) {
+    return {
+      type: 0,
+      code: asset.getCode(),
+    };
+  }
+  return {
+    type: asset.getAssetType() === "credit_alphanum4" ? 1 : 2,
+    code: asset.getCode(),
+    issuer: asset.getIssuer(),
+  };
+};
+
+/**
+ * Transforms amount from decimals (lumens) to integer (stroop)
+ */
+const transformAmount = (amount: string) =>
+  new BigNumber(amount).times(10000000).toString();
+
+/**
+ * Transforms StellarSdk.Operation.type to TrezorConnect.StellarTransaction
+ * Operation.type
+ */
+const transformType = (type: string) => {
+  switch (type) {
+    case "pathPaymentStrictReceive":
+      // case 'pathPaymentStrictSend':
+      return "pathPayment";
+
+    case "createPassiveSellOffer":
+      return "createPassiveOffer";
+
+    case "manageSellOffer":
+      // case 'manageBuyOffer':
+      return "manageOffer";
+    default:
+      return type;
+  }
+};
+
+const transformMemo = (memo: { type: MemoType; value: MemoValue }) => {
```

Comment:
this does not work, and the linked file from Connect has something very different:
https://github.com/trezor/connect/blob/develop/src/js/plugins/stellar/plugin.js#L86-L101

see also https://github.com/trezor/trezor-firmware/issues/1694

Reply:
Thanks, @matejcik, for pointing this out! 🙌  The fix is live now.